src/main/java/jenkins/plugins/nodejs/tools/pathresolvers/LatestInstallerPathResolver.java

```diff
@@ -94,14 +98,16 @@ public String resolvePathFor(String version, Platform platform, CPU cpu) {
         case i386:
             if (platform == Platform.OSX && nodeVersion.compareTo(new NodeJSVersion(4, 0, 0)) >= 0 //
                     || ((platform == Platform.SUNOS || platform == Platform.LINUX)
-                            && nodeVersion.compareTo(new NodeJSVersion(10, 0, 0)) >= 0)) {
+                            && nodeVersion.compareTo(new NodeJSVersion(10, 0, 0)) >= 0) //
+                    || platform == Platform.AIX) {
```

Comment:
When could be the case of AIX installed o a i386 CPU?

Reply:
Hello, during the test done in the build step, I was facing some issues, so I wanted to be as secured as possible. This is same comment for 3 cases.

Reply:
which issues?

Reply:
I have just done a new build without this useless cases, but I get following error.
(I'm a newbie in git and so on, so I may have done some bad things)

`
[INFO] Running jenkins.plugins.nodejs.tools.InstallerPathResolversTest
Exception in thread "Thread-4" java.lang.OutOfMemoryError: Java heap space
        at java.util.Arrays.copyOfRange(Arrays.java:3664)
        at java.lang.String.<init>(String.java:207)
        at java.lang.StringBuilder.toString(StringBuilder.java:407)
        at java.io.BufferedReader.readLine(BufferedReader.java:359)
        at java.io.BufferedReader.readLine(BufferedReader.java:389)
        at org.apache.maven.surefire.shade.common.org.apache.maven.shared.utils.cli.StreamPumper.run(StreamPumper.java:89)
`

`
[ERROR] Crashed tests:
[ERROR] jenkins.plugins.nodejs.tools.InstallerPathResolversTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C "E:\Tools\JDK8u261\jre\bin\java -Xms768M -Xmx768M -Djava.awt.headless=true -XX:+HeapDumpOnOutOfMemoryError -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar C:\Users\nct\AppData\Local\Temp\surefire2614204000096579939\surefirebooter8457310949635492458.jar C:\Users\nct\AppData\Local\Temp\surefire2614204000096579939 2021-01-12T10-00-59_579-jvmRun1 surefire8617497962086721327tmp surefire_0708089835637974812tmp"
[ERROR] Process Exit Code: 0
[ERROR] Crashed tests:
[ERROR] jenkins.plugins.nodejs.tools.InstallerPathResolversTest
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:690)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:248)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1217)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1063)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:889)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR]
[ERROR] -> [Help 1]
`

Reply:
this is a test issue, or your maven run has no enought memory or there is a loop in the test or the test input file is become too big

Reply:
Ok.
So I will try to fix it on my side without this useless test cases.

Reply:
On my PC no issues.

Reply:
Is there a recommended JVM to use ?
I was using Oracle JDK 8 u261

Reply:
no, that's ok. Memory should be 512M. Anyway let the build system say if it's ok.

Reply:
Seems to be OK with AdoptOpenJDK hotspot jdk8u242-b08
So I did expected fixes.
Hope my git command are correct